apis/apps/v1alpha1/nebulacluster_metad.go

```diff
@@ -101,10 +101,16 @@ func (c *metadComponent) GetDataStorageClass() *string {
 }
 
 func (c *metadComponent) GetLogStorageResources() *corev1.ResourceRequirements {
+	if c.nc.Spec.Metad.LogVolumeClaim == nil {
+		return nil
+	}
 	return c.nc.Spec.Metad.LogVolumeClaim.Resources.DeepCopy()
 }
 
 func (c *metadComponent) GetDataStorageResources() (*corev1.ResourceRequirements, error) {
+	if c.nc.Spec.Metad.DataVolumeClaim == nil {
```

Comment:
I understand that the request is `LogVolumeClaim` can be nil. But why `DataVolumeClaim` can be nil too?

Reply:
prevent panic when the DataVolumeClaim does not exist